python/nn4k/nn4k/invoker/openai.py

```diff
@@ -35,41 +36,107 @@ def __init__(self, nn_config: dict):
         self.openai_api_base = get_string_field(
             self.init_args, NN_OPENAI_API_BASE_KEY, NN_OPENAI_API_BASE_TEXT
         )
-        self.openai_max_tokens = get_positive_int_field(
-            self.init_args, NN_OPENAI_MAX_TOKENS_KEY, NN_OPENAI_MAX_TOKENS_TEXT
-        )
+        self.openai_max_tokens = self.init_args.get(NN_OPENAI_MAX_TOKENS_KEY)
+        if self.openai_max_tokens is not None:
+            self.openai_max_tokens = get_positive_int_field(
+                self.init_args, NN_OPENAI_MAX_TOKENS_KEY, NN_OPENAI_MAX_TOKENS_TEXT
+            )
+        self.openai_organization = self.init_args.get(NN_OPENAI_ORGANIZATION_KEY)
+        if self.openai_organization is not None:
+            self.openai_organization = get_string_field(
+                self.init_args, NN_OPENAI_ORGANIZATION_KEY, NN_OPENAI_ORGANIZATION_TEXT
+            )
 
-        openai.api_key = self.openai_api_key
-        openai.api_base = self.openai_api_base
+        if self._is_legacy_openai_api:
+            self.client = None
+            openai.api_key = self.openai_api_key
+            openai.api_base = self.openai_api_base
+            openai.organization = self.openai_organization
+        else:
+            self.client = openai.OpenAI(
+                api_key=self.openai_api_key,
+                base_url=self.openai_api_base,
+                organization=self.openai_organization,
+            )
 
     @classmethod
     def from_config(cls, nn_config: dict) -> "OpenAIInvoker":
         invoker = cls(nn_config)
         return invoker
 
+    @property
+    def _is_legacy_openai_api(self):
+        import openai
+
+        return openai.__version__.startswith("0.")
+
     def _create_prompt(self, input, **kwargs):
         if isinstance(input, list):
             prompt = input
         else:
-            prompt = [input]
+            prompt = [{"role": "user", "content": input}]
         return prompt
 
+    def _create_completion(self, input, prompt, max_output_length, **kwargs):
+        import openai
+
+        if self._is_legacy_openai_api:
+            completion = openai.ChatCompletion.create(
+                model=self.openai_model_name,
```

Comment:
add spaces on both side of `=`

Reply:
I tried, the Python code formatter **black** will format ``func(a = 1)`` back to ``func(a=1)``.